wg-api/spec-documents/clipboard-rearchitecture.md

```diff
@@ -0,0 +1,105 @@
+
+At this point, it becomes clear that it is in users' and maintainers' best interest to reconsider how we approach the `clipboard` module - breaking changes might be difficult in the short term but provide more consistency and functionality as we look to the future of the module.
+
+## API Design
+
+The largest consideration for changes to the `clipboard` module is to what degree we should choose to align with the W3C specification. As we consider this, we should consider what needs or use cases might exist for Electron uses which necessarily wouldn't exist for Chrome users, and how we would plan to account for that in our new design.
+
+### Potential Options
+
+1. Strict alignment with W3C specification - no additional APIs exposed.
+2. Alignment with W3C where possible, with some additional APIs exposed to handle desktop use cases which aren’t addressed by the specification.
+3. Leave API as-is and continue to adapt it as required by upstream changes by Chromium development.
+ 
+### Preferred Approach/Solution
+
+Alignment with W3C where possible, with some additional APIs exposed to handle desktop use cases which aren’t considered by the specification.
+
+**APIs to Preserve**
+* `clipboard.readBookmark()`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.writeBookmark(title, url[, type])`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.readFindText()` _macOS_
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.writeFindText(text)`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.clear([type])`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.readText([type])`
+  * Already implemented per specification.
+* `clipboard.writeText(text[, type])`
+  * Already implemented per specification.
+
+**APIs to Remove**
+* `clipboard.availableFormats([type])`
+  * Superseded by `clipboard.read` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.has(format[, type])` 
+  * Superseded by `clipboard.read` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.readBuffer(format)`
+  * Superseded by `clipboard.read`[Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.writeBuffer(format, buffer[, type])`
+  * Superseded by `clipboard.write` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write)
+* `clipboard.readHTML()`
+  * Superseded by `clipboard.read` with `text/html` MIME type
+* `clipboard.writeHTML(markup[, type])`
+  * Superseded by `clipboard.write` with `text/html` MIME type
```

Comment:
Does this have the same semantics as clipboard.write with 'text/html'? The docs for these two functions imply there's an internal transform, e.g. the addition of a `<meta charset='utf-8'>`

This question might not matter since this proposal is a breaking change anyway, but I'm curious so I'll ask anyway :smile:

Reply:
@ckerr I just checked.  Electron calls the underlying Chromium APIs, so that `<meta charset='utf-8'>` gets added by chromium and will continue to do so under the new implementation.  You can test it yourself at https://fuzzy-wobbly-splash.glitch.me/